slider/index.js

```diff
@@ -0,0 +1,170 @@
+class IRNMNSlider extends HTMLElement {
+
+    CLASSNAMES = [];
+
+    constructor() {
+        super();
+        this.CLASSNAMES = this.selectors;
+        this.currentSlide = 1; // Start from the first visible slide after cloning        
+    }
+
+    /**
+     * Get the selectors from the attribute
+     * to be used as classnames obj
+     * 
+     * @returns {object} - The selectors object
+     */
+    get selectors() {
+        let selectors = this.getAttribute('selectors');
+        selectors = JSON.parse(selectors);
+        let classnames = [];
+        // replace classnames with selectors
+        for (let key in selectors) {
+            classnames[key.toUpperCase()] = selectors[key];
+        }
+
+        return classnames;
+    }
+
+    connectedCallback() {
+        this.initSlider();
+    }
+
+    initSlider() {
+
+        const swipeContainer = this.querySelector(this.CLASSNAMES.SWIPE_CONTAINER);
+        if ( !swipeContainer ) {
+            console.error('Swipe container not found');
+            return;
+        }
+        const slides = swipeContainer.querySelectorAll(this.CLASSNAMES.SLIDES);
+        const totalSlides = slides.length;
+
+        if (this.dataset.sliderInitialized === "true") {
+            return;
+        }
+        this.dataset.sliderInitialized = "true";
+
+        /**
+         * Handle slider loopping
+         */
+        const firstClone = slides[0].cloneNode(true);
```

Comment:
@samuelsilvapt Are you handling  a scenario where only one image is provided? Would this logic still make sense?

Reply:
I added a new method to handle the one slider only fallback - following swiper logic, renderSingleSlide removes the navigation if exists and display the single slide normally